### PR TITLE
Issue #2818111 - Ensure order items show up in Cart block

### PR DESCRIPTION
**Pull request by steveoliver**
Failing test shows that order items are not being displayed in cart block.

### CODE REVIEW

File: modules/cart/tests/src/Functional/CartBlockTest.php [hunk 1]
```diff
@@ -41,4 +41,17 @@ public function testCountText() {
     $this->assertSession()->pageTextNotContains('3 items');
   }
 
+  /**
+   * Tests the order items display.
+   */
+  public function testOrderItemsDisplay() {
+    $this->drupalGet('<front>');
```
Comment (bojanz) [inline modules/cart/tests/src/Functional/CartBlockTest.php]: How is this test method different from the one above? They both add an entity to cart and assert the count. Which makes me think the previous one should have caught the bug?

Comment (steveoliver) [reply modules/cart/tests/src/Functional/CartBlockTest.php]: Test above only tests for the display of the 'X item[s]' text -- It's different than the order items list (view), which wasn't tested above.

File: modules/cart/tests/src/Functional/CartBlockTest.php [hunk 2]
```diff
@@ -41,4 +41,17 @@ public function testCountText() {
...
+    $this->assertSession()->pageTextContains('0 items');
+
+    $this->cartManager->addEntity($this->cart, $this->variation);
+    $this->drupalGet('<front>');
+    $this->assertSession()->pageTextContains('1 item');
+    $this->assertSession()->pageTextContains($this->variation->getOrderItemTitle());
```
Comment (steveoliver) [inline modules/cart/tests/src/Functional/CartBlockTest.php]: We could remove the assertions for X item[s] -- they are indeed exact dupes of the first test.  This line 54 checking OrderItemTitle is the previous uncaught fail.

**Comment by steveoliver** [comment]
Really weird the first commit test passes on Travis. :/

File: modules/cart/tests/src/Functional/CartBlockTest.php (hunk 1 above)
Comment (mglaman) [reply modules/cart/tests/src/Functional/CartBlockTest.php]: Could we have the test just check count + title/cart block view?

Comment (steveoliver) [reply modules/cart/tests/src/Functional/CartBlockTest.php]: @mglaman, I combined the tests in the latest commit.

**Review by steveoliver — COMMENTED** [review]
As suspected, this (original state of the view) makes the new cart order item test fail.

**Pull request merged**